This is the log on a German system. The file names are different/localized.

```
[2022-03-22 17:48:39.3045] [ERROR] [C:\a\_work\1\s\src\modules\launcher\Plugins\Microsoft.Plugin.Program\Programs\Win32Program.cs::502]
-------------------------- Begin exception --------------------------
Message: 	Error status: UNKNOWN
		Program path: C:\Users\Heiko\Desktop\Neue Verknüpfung.lnk
		Exception thrown in called method: Void Load(System.String, Int32)
		Possible interpretation of the error: |An unexpected error occurred in the calling method LnkProgram at C:\Users\Heiko\Desktop\Neue Verknüpfung.lnk
		Possible resolution: Not yet known

Exception full name  : System.IO.FileLoadException
Exception message    : Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird. (0x80070020)
Exception stack trace:
   at System.Runtime.InteropServices.ComTypes.IPersistFile.Load(String pszFileName, Int32 dwMode)
   at Microsoft.Plugin.Program.Programs.ShellLinkHelper.RetrieveTargetPath(String path)
   at Microsoft.Plugin.Program.Programs.Win32Program.LnkProgram(String path)
Exception source     : System.Private.CoreLib
Exception target site: Void Load(System.String, Int32)
Exception HResult    : -2147024864
-------------------------- End exception --------------------------


[2022-03-22 17:48:39.3045] [ERROR] [C:\a\_work\1\s\src\modules\launcher\Wox.Infrastructure\Storage\ListRepository`1.cs::62]
Item Not Found <~eue Verknüpfung.tmp>
```

--- a/src/modules/launcher/Plugins/Microsoft.Plugin.Program.UnitTests/Storage/Win32ProgramRepositoryTest.cs
+++ b/src/modules/launcher/Plugins/Microsoft.Plugin.Program.UnitTests/Storage/Win32ProgramRepositoryTest.cs
@@ -352,10 +352,11 @@ namespace Microsoft.Plugin.Program.UnitTests.Storage
 
             string oldFullPath = directory + "\\" + oldpath;
             string fullPath = directory + "\\" + path;
+            string linkingTo = Directory.GetCurrentDirectory();
 
             // ShellLinkHelper must be mocked for lnk applications
             var mockShellLink = new Mock<IShellLinkHelper>();
-            mockShellLink.Setup(m => m.RetrieveTargetPath(It.IsAny<string>())).Returns(string.Empty);
+            mockShellLink.Setup(m => m.RetrieveTargetPath(It.IsAny<string>())).Returns(linkingTo);
             Win32Program.ShellLinkHelper = mockShellLink.Object;
 
             // old item and new item are the actual items when they are in existence
@@ -363,14 +364,14 @@ namespace Microsoft.Plugin.Program.UnitTests.Storage
             {
                 Name = "oldpath",
                 ExecutableName = oldpath,
-                FullPath = fullPath,
+                FullPath = linkingTo,
             };
 
             Win32Program newitem = new Win32Program
             {
                 Name = "path",
                 ExecutableName = path,
-                FullPath = fullPath,
+                FullPath = linkingTo,
             };
 
             win32ProgramRepository.Add(olditem);

--- a/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs
+++ b/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs
@@ -491,8 +491,18 @@ namespace Microsoft.Plugin.Program.Programs
                         }
                     }
                 }
+                else
+                {
+                    // If the link points nowhere, consider it invalid.
+                    return InvalidProgram;
+                }
 
                 return program;
+            }
+            catch (System.IO.FileLoadException e)
+            {
+                ProgramLogger.Warn($"|Couldn't load the link file at {path}. This might be caused by a new link being created and locked by the OS.", e, MethodBase.GetCurrentMethod().DeclaringType, path);
+                return InvalidProgram;
             }
 
             // Only do a catch all in production. This is so make developer aware of any unhandled exception and add the exception handling in.

--- a/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs
+++ b/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs
@@ -501,7 +501,7 @@ namespace Microsoft.Plugin.Program.Programs
             }
             catch (System.IO.FileLoadException e)
             {
-                ProgramLogger.Warn($"|Couldn't load the link file at {path}. This might be caused by a new link being created and locked by the OS.", e, MethodBase.GetCurrentMethod().DeclaringType, path);
+                ProgramLogger.Warn($"Couldn't load the link file at {path}. This might be caused by a new link being created and locked by the OS.", e, MethodBase.GetCurrentMethod().DeclaringType, path);
                 return InvalidProgram;
             }
 

--- a/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/Win32ProgramRepository.cs
+++ b/src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/Win32ProgramRepository.cs
@@ -99,7 +99,7 @@ namespace Microsoft.Plugin.Program.Storage
             string newPath = e.FullPath;
 
             string extension = Path.GetExtension(newPath);
-            Win32Program.ApplicationType appType = Win32Program.GetAppTypeFromPath(newPath);
+            Win32Program.ApplicationType oldAppType = Win32Program.GetAppTypeFromPath(oldPath);
             Programs.Win32Program newApp = Win32Program.GetAppFromPath(newPath);
             Programs.Win32Program oldApp = null;
 
@@ -109,11 +109,7 @@ namespace Microsoft.Plugin.Program.Storage
             // This situation is not encountered for other application types because the fullPath is the path itself, instead of being computed by using the path to the app.
             try
             {
-                if (appType == Win32Program.ApplicationType.ShortcutApplication)
-                {
-                    oldApp = new Win32Program() { Name = Path.GetFileNameWithoutExtension(e.OldName), ExecutableName = Path.GetFileName(e.OldName), FullPath = newApp.FullPath };
-                }
-                else if (appType == Win32Program.ApplicationType.InternetShortcutApplication)
+                if (oldAppType == Win32Program.ApplicationType.ShortcutApplication || oldAppType == Win32Program.ApplicationType.InternetShortcutApplication)
                 {
                     oldApp = new Win32Program() { Name = Path.GetFileNameWithoutExtension(e.OldName), ExecutableName = Path.GetFileName(e.OldName), FullPath = newApp?.FullPath ?? oldPath };
                 }
@@ -160,6 +156,11 @@ namespace Microsoft.Plugin.Program.Storage
                 if (extension.Equals(LnkExtension, StringComparison.OrdinalIgnoreCase))
                 {
                     app = GetAppWithSameLnkResolvedPath(path);
+                    if (app == null)
+                    {
+                        // Cancelled links won't be have a resolved path.
+                        app = GetAppWithSameNameAndExecutable(Path.GetFileNameWithoutExtension(path), Path.GetFileName(path));
+                    }
                 }
                 else if (extension.Equals(UrlExtension, StringComparison.OrdinalIgnoreCase))
                 {

